packages/components/src/unit-control/index.native.js

```diff
@@ -115,19 +116,20 @@ function UnitControl( {
 	};
 
 	const renderUnitPicker = useCallback( () => {
+		if ( hasUnits( units ) ) {
+			return null;
+		}
 		return (
 			<View style={ styles.unitMenu } ref={ anchorNodeRef }>
 				{ renderUnitButton }
-				{ hasUnits( units ) && units?.length > 1 ? (
```

Comment:
I think we need to keep the `units?.length > 1`

Introduced from this [PR](https://github.com/WordPress/gutenberg/pull/33641).

Reply:
for sure make sense to me!